synapse/handlers/sliding_sync.py

```diff
@@ -275,12 +275,6 @@ async def get_sync_room_ids_for_user(
             instance_map=immutabledict(instance_to_max_stream_ordering_map),
         )
 
-        # If our `to_token` is already the same or ahead of the latest room membership
-        # for the user, we can just straight-up return the room list (nothing has
-        # changed)
-        if membership_snapshot_token.is_before_or_eq(to_token.room_key):
-            return sync_room_id_set
```

Comment:
We could have also fixed this by making sure `from_token is None` in this condition:

```suggestion
        if from_token is None and membership_snapshot_token.is_before_or_eq(to_token.room_key):
            return sync_room_id_set
```

But the way the PR is now, puts the condition next to the relevant fixups and we can avoid the extra membership lookup if we only need to do the `newly_left` fixup "2)"